cli/cmd/deploy/deploy.go

```diff
@@ -473,6 +486,63 @@ func extractGitRemote(projectPath string) (string, error) {
 	return gitutil.RemotesToGithubURL(remotes)
 }
 
+func isLocalInSyncWithRemote(repoPath, branch string) error {
```

Comment:
1. Move to `package gitutil`
2. Seems like it's overloading `error` to provide a status. Maybe refactor to `gitutil.SyncStatus(path, branch string) (SyncStatus, error)`?